src/model/project.rs

```diff
@@ -0,0 +1,42 @@
+use serde::Deserialize;
+
+#[derive(Deserialize, Debug, Clone)]
+pub struct Project {
+    id: String,
+    name: String,
+    parent_id: Option<String>,
+
+    #[serde(skip)]
+    depth: usize,
+}
+
+impl Project {
+    pub fn new(id: String, name: String, parent_id: Option<String>) -> Project {
+        Project {
+            id,
+            name,
+            parent_id,
+            depth: 0,
+        }
+    }
```

Comment:
This function isn't used right now since every object is constructed through serde. Let's remove this constructor.

Reply:
Done.